[Doc] Update the documentation about Docker images and the environment variables which configure it (#2135)
strimzi/strimzi-kafka-operator: documentation/modules/ref-configuring-container-images.adoc

* `KafkaConnectS2I.spec`
* `KafkaBridge.spec`

> == Configuring the `image` property for Kafka, Kafka Connect and Kafka Mirror Maker

Reviewer: ```suggestion
== Configuring the `image` property for Kafka, Kafka Connect, and Kafka Mirror Maker
```

> 

Reviewer: Please can you re-word the lead-in sentence for the bullet list.

EXISTING: Container image which should be used for given components can be specified using the `image` property in:

NEW: You can specify which container image to use for each component using the `image` property in the following resources:

> Kafka, Kafka Connect (including Kafka Connect with S2I support), and Kafka Mirror Maker support multiple versions of Kafka, each requiring the own image.

Reviewer: ```suggestion
Kafka, Kafka Connect (including Kafka Connect with S2I support), and Kafka Mirror Maker support multiple versions of Kafka. Each component requires its own image.
```

> The default images for the different versions are configured in the following environment variables:

Reviewer: ```suggestion
The default images for the different Kafka versions are configured in the following environment variables:
```

* `STRIMZI_KAFKA_IMAGES`
* `STRIMZI_KAFKA_CONNECT_IMAGES`
* `STRIMZI_KAFKA_CONNECT_S2I_IMAGES`
* `STRIMZI_KAFKA_MIRROR_MAKER_IMAGES`

> The environment variables contain mapping between Kafka versions and the corresponding images.

Reviewer: ```suggestion
These environment variables contain mappings between the Kafka versions and their corresponding images.
```

> The mappings are used in combination with the `image` and `version` properties as follows:

Reviewer: ```suggestion
The mappings are used together with the `image` and `version` properties:
```

* If neither `image` nor `version` are given in the custom resource then the `version` will default to the Cluster Operator's default Kafka version, and the image will be the one corresponding to this version in the environment variable.

> * If `image` is given but `version` is not then the given image will be used and the `version` will be assumed to be the Cluster Operator's default Kafka version.

Reviewer: ```suggestion
* If `image` is given but `version` is not, then the given image is used and the `version` is assumed to be the Cluster Operator's default Kafka version.
```

> * If `version` is given but `image` is not then image will be the one corresponding to this version in the environment variable.

Reviewer: ```suggestion
* If `version` is given but `image` is not, then the image that corresponds to the given version in the environment variable is used.
```

> * If both `version` and `image` are given the given image will be used, and it will be assumed to contain a Kafka with the given version.

Reviewer: ```suggestion
* If both `version` and `image` are given, then the given image is used. The image is assumed to contain a Kafka image with the given version.
```

The `image` and `version` for the different components can be configured in the following properties:

* For Kafka in `spec.kafka.image` and `spec.kafka.version`.
* For Kafka Connect, Kafka Connect S2I, and Kafka Mirror Maker in `spec.image` and `spec.version`.

> WARNING: It is best to provide just `version` and leave the `image` property unspecified.

Reviewer: ```suggestion
WARNING: It is recommended to provide only the `version` and leave the `image` property unspecified.
```

> This reduces the chances of making a mistake in configuring the custom resource. 

Reviewer: ```suggestion
This reduces the chance of making a mistake when configuring the custom resource. 
```

> If you need to change the images used for different versions of Kafka, it is better to configure the Cluster Operator's environment variables.

Reviewer: ```suggestion
If you need to change the images used for different versions of Kafka, it is preferable to configure the Cluster Operator's environment variables.
```
